app/domain/authentication/authn_gcp/validate_resource_restrictions.rb

```diff
@@ -0,0 +1,53 @@
+module Authentication
+  module AuthnGcp
+
+    # This class is responsible of validating resource restrictions that configured on a Conjur host or user
+    # against google JWT token.
+    ValidateResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions:                ExtractResourceRestrictions.new,
+        validate_resource_restrictions_configuration: ValidateResourceRestrictionsConfiguration.new,
+        validate_resource_restrictions_match_jwt:     ValidateResourceRestrictionsMatchJWT.new,
+        logger:                                       Rails.logger
+      },
+      inputs:       [:authenticator_input]
```

Comment:
i think that it's not very good to pass around the `authenticator_input` object. instead, it will be better to state `:account, :username, :credentials` so that it's clear from the class signature what the input is.

wdyt?

Reply:
OK